pkg/image/registry/imagestreamtag/rest.go

```diff
@@ -93,12 +135,54 @@ func (r *REST) Get(ctx kapi.Context, id string) (runtime.Object, error) {
 			Namespace:         kapi.NamespaceValue(ctx),
 			Name:              id,
 			CreationTimestamp: event.Created,
+			Annotations:       istagAnnotations,
+
+			// resource version is logically the later of image or imagestream
+			// with numbers, concatenating the two will alway produce a number larger than either one
+			// with strings, it will produce something unique.
+			ResourceVersion: stream.ResourceVersion + image.ResourceVersion,
```

Comment:
> Edit only affects the image stream, not the image.

But the decision to modify is based on looking at an entire object, not just the bit you're mutating.